### PR TITLE
examples: add `transform` example

### PR DESCRIPTION
**Pull request by yoshuawuyts**
Adds a simple example showing how to read a file, transform it and then pipe it back out. Took me a while to figure out how `csv-parser` and `csv-write-stream` can work together. Figured this might help other stream beginners get started. Thanks!

### CODE REVIEW

File: examples/transform.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  .pipe(through.obj(transform))
+  .pipe(write())
+  .pipe(process.stdout)
+
+// Prepend all chunks with `value: `.
+// @param {Object} chunk
+// @param {String} encoding
+// @param {Function} callback
+function transform (chunk, enc, cb) {
+  this.push('value: ' + chunk)
```
Comment (mafintosh) [inline examples/transform.js]: should you iterate over the keys in `chunk` and prepend the values?

```
Object.keys(chunk).forEach(function(k) {
  chunk[k] = 'value: ' + chunk[k]
})
```

?

Comment (yoshuawuyts) [inline examples/transform.js]: :100:

**Comment by yoshuawuyts** [comment]
Fixed with feedback. Some `require()` statements were also broken. Everything works as expected now:
```sh
~/src/mafintosh/csv-parser example*
❯ node ./examples/transform.js
a,b,c
value: 1,value: 2,value: 3
```

**Comment by mafintosh** [comment]
Nice. Thanks for the contribution

**Pull request merged**